### PR TITLE
Handle missing packages for FFT baked collisions

### PR DESCRIPTION
**Pull request by magenta404**
Wraps burst attributes in defines as suggested by @kpietraszko. If the burst package is missing, it compiles but baking crashes Unity.

Also wraps baking classes with defines and wherever they are used. This seems like the simplest approach. I don't think wrapping a serialised property is really that much of an issue for this use case now that I think about it as losing the reference would be an edge case as it would require a user to remove the package and then serialise the reference holder.

Lastly, adds validation and fix functions to add missing packages.

ping @huwb in case you don't get a notification.

### CODE REVIEW

File: crest/Assets/Crest/Crest/Scripts/LodData/Settings/SimSettingsAnimatedWaves.cs
Comment (kpietraszko) [inline crest/Assets/Crest/Crest/Scripts/LodData/Settings/SimSettingsAnimatedWaves.cs]: Wrong package ID I think?

Comment (magenta404) [reply crest/Assets/Crest/Crest/Scripts/LodData/Settings/SimSettingsAnimatedWaves.cs]: Oops. Thanks.

**Comment by huwb** [comment]
I didn't get any notification at all! Will take a look

**Comment by huwb** [comment]
> If the burst package is missing, it compiles but baking crashes Unity

Does it? Do we need to disable the bake button if burst is absent then?

File: crest/Assets/Crest/Crest/Scripts/Collision/CollProviderBakedFFT.cs
Comment (huwb) [inline crest/Assets/Crest/Crest/Scripts/Collision/CollProviderBakedFFT.cs]: I wonder if this should be more explicit, maybe CREST_UNITY_MATHEMATICS. Not a deal breaker for me though.

**Review by huwb — APPROVED** [review]
Minor feedback but will approve so you can merge when you're happy

**Comment by huwb** [comment]
> Lastly, adds validation and fix functions to add missing packages.

Great stuff

File: crest/Assets/Crest/Crest/Scripts/Collision/CollProviderBakedFFT.cs
Comment (magenta404) [reply crest/Assets/Crest/Crest/Scripts/Collision/CollProviderBakedFFT.cs]: Sure. Done for burst too to be consistent.

**Comment by magenta404** [comment]
> > If the burst package is missing, it compiles but baking crashes Unity
> 
> Does it? Do we need to disable the bake button if burst is absent then?

I tried this again and it baked with no burst, but crashed when I pressed play using the baked data. I baked another one with burst added, then disabled burst, then pressed play and it didn't crash. We should disable button if it cannot be solved.

**Comment by magenta404** [comment]
It's ready to merge unless you want me to disable the button too @huwb? I cannot merge it so that might be @kpietraszko who needs to.

**Comment by huwb** [comment]
Hmm maybe it should merge as is. I guess the crash can be considered separately

**Comment by huwb** [comment]
I'll go ahead and merge to keep things ticking over, hope thats ok @kpietraszko

**Pull request merged**
